segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -349,12 +351,15 @@ public boolean isOffline() {
         return this.metadataStore.getOrAssignSegmentId(streamSegmentName, timer.getRemaining(),
                 streamSegmentId -> {
                     StreamSegmentAppendOperation operation = new StreamSegmentAppendOperation(streamSegmentId, data, attributeUpdates);
-                    return processAttributeUpdaterOperation(operation, timer);
+                    return processAttributeUpdaterOperation(operation, timer).thenApply(v -> {
+                        SegmentMetadata segmentMetadata = this.metadata.getStreamSegmentMetadata(streamSegmentId);
```

Comment:
No need for this. `StreamSegmentAppendOperation` will contain the offset where this was added. just return `operation.getStreamSegmentOffset()`